tiledb/sm/c_api/tiledb.cc

```diff
@@ -935,6 +935,22 @@ int32_t tiledb_config_unset(
   return TILEDB_OK;
 }
 
+int32_t tiledb_config_compare(
+    tiledb_config_t* lhs, tiledb_config_t* rhs, uint8_t* equal) {
+  if (equal == 0)
```

Comment:
check against `nullptr` instead of `0`. additionally, log a status error here, similar to how we do elsewhere in the C API with `LOG_STATUS`.